analytics-datastore-clickhouse/swarm.sh

```diff
@@ -1,107 +1,113 @@
 #!/bin/bash
 
-readonly ACTION=$1
-readonly MODE=$2
-
-COMPOSE_FILE_PATH=$(
-  cd "$(dirname "${BASH_SOURCE[0]}")" || exit
-  pwd -P
-)
-readonly COMPOSE_FILE_PATH
-
-ROOT_PATH="${COMPOSE_FILE_PATH}/.."
-readonly ROOT_PATH
-
-. "${ROOT_PATH}/utils/config-utils.sh"
-. "${ROOT_PATH}/utils/docker-utils.sh"
-. "${ROOT_PATH}/utils/log.sh"
-
-main() {
-  if [[ "${ACTION}" == "init" ]] || [[ "${ACTION}" == "up" ]]; then
-    local clickhouse_dev_compose_param=""
-
-    if [[ "${STATEFUL_NODES}" == "cluster" ]]; then
-      log info "Running Analytics Datastore Clickhouse package in Cluster node mode"
-
-      log info "Setting config digests"
-      config::set_config_digests "${COMPOSE_FILE_PATH}/docker-compose.cluster.yml"
-
-      if [[ "${MODE}" == "dev" ]]; then
-        log info "Running Analytics Datastore Clickhouse package in DEV mode"
-        clickhouse_dev_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose.cluster.dev.yml"
-      fi
+declare ACTION=""
+declare MODE=""
+declare COMPOSE_FILE_PATH=""
+declare ROOT_PATH=""
+declare nodes_mode=""
+declare service_names=()
+
+function init_vars() {
+  ACTION=$1
+  MODE=$2
+
+  COMPOSE_FILE_PATH=$(
+    cd "$(dirname "${BASH_SOURCE[0]}")" || exit
+    pwd -P
+  )
+
+  ROOT_PATH="${COMPOSE_FILE_PATH}/.."
+
+  if [[ "${NODE_MODE}" == "cluster" ]]; then
+    nodes_mode=".${NODE_MODE}"
+    service_names=(
+      "analytics-datastore-clickhouse-01"
+      "analytics-datastore-clickhouse-02"
+      "analytics-datastore-clickhouse-03"
+      "analytics-datastore-clickhouse-04"
+    )
```

Comment:
let's do a "for i in {1...4}"  loop

Reply:
cool done in [201d509](https://github.com/jembi/platform/pull/188/commits/201d50983863521fe4ed85a031505b5d1e4a8e87)